dotCMS/src/main/java/com/dotcms/storage/DataBaseStoragePersistenceAPIImpl.java

```diff
@@ -0,0 +1,623 @@
+import java.util.Optional;
+import java.util.concurrent.Future;
+
+/**
+ * Represents a Storage on the database It supports big files since provides the ability to split
+ * fat object in smaller pieces, see {@link FileByteSplitter} and {@link com.dotcms.util.FileJoiner}
+ * to get more details about the process
+ *
+ * @author jsanca
+ */
+public class DataBaseStoragePersistenceAPIImpl implements StoragePersistenceAPI {
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    protected Connection getConnection() {
+
+        final String jdbcPool = Config
+                .getStringProperty("DATABASE_STORAGE_JDBC_POOL_NAME", StringPool.BLANK);
+        final boolean isExternalPool = UtilMethods.isSet(jdbcPool);
+
+        return isExternalPool ?
+                DbConnectionFactory.getConnection(jdbcPool) :
+                DbConnectionFactory.getConnection();
+    }
+
+    protected void wrapCloseConnection(final VoidDelegate voidDelegate) {
+
+        final String jdbcPool = Config
+                .getStringProperty("DATABASE_STORAGE_JDBC_POOL_NAME", StringPool.BLANK);
+        final boolean isExternalPool = UtilMethods.isSet(jdbcPool);
+
+        if (isExternalPool) {
+
+            wrapExternalCloseConnection(voidDelegate, jdbcPool);
+        } else {
+
+            wrapLocalCloseConnection(voidDelegate);
+        }
+    }
+
+    private void wrapLocalCloseConnection(final VoidDelegate voidDelegate) {
+
+        final boolean isNewConnection = !DbConnectionFactory.connectionExists();
+
+        try {
+
+            voidDelegate.execute();
+        } catch (DotSecurityException | DotDataException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new DotRuntimeException(e);
+        } finally {
+
+            if (isNewConnection) {
+
+                DbConnectionFactory.closeSilently();
+            }
+        }
+    }
+
+    private void wrapExternalCloseConnection(final VoidDelegate voidDelegate,
+            final String jdbcPool) {
+
+        Connection connection = null;
+        try {
+
+            connection = DbConnectionFactory.getConnection(jdbcPool);
+            voidDelegate.execute();
+        } catch (DotSecurityException | DotDataException e) {
+
+            Logger.error(this, e.getMessage(), e);
+            throw new DotRuntimeException(e);
+        } finally {
+
+            CloseUtils.closeQuietly(connection);
+        }
+    }
+
+    protected <T> T wrapInTransaction(final ReturnableDelegate<T> delegate) {
+
+        final String jdbcPool = Config
+                .getStringProperty("DATABASE_STORAGE_JDBC_POOL_NAME", StringPool.BLANK);
+        final boolean isExternalPool = UtilMethods.isSet(jdbcPool);
+
+        if (isExternalPool) {
+
+            return this.wrapInExternalTransaction(delegate, jdbcPool);
+        } else {
+
+            try {
+
+                return LocalTransaction.wrapReturnWithListeners(delegate);
+            } catch (Exception e) {
```

Comment:
Codacy found an issue: [Avoid catching generic exceptions such as NullPointerException, RuntimeException, Exception in try-catch block](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)